Composer/packages/server/src/models/storage/localDiskStorage.ts

```diff
@@ -61,8 +63,12 @@ export class LocalDiskStorage implements IFileStorage {
     await mkDir(path, options);
   }
 
-  async rmDir(path: string): Promise<void> {
-    await rmDir(path);
+  async rmDir(path: string, force?: boolean): Promise<void> {
+    if (force) {
+      await forceRmDir(path);
+    } else {
+      await rmDir(path);
+    }
   }
```

Comment:
`rmdir` and `rm -rf` are different operations. `rm -rf` is not "force rmdir to happen", it's "remove a directory and everything inside it", and it ought to be its own method here. I'm fine with using `rimraf` as a way to implement it, but the boolean switch to choose between the two is not the way I'd go. `rmdir` should stay as pure `rmdir`. As a side benefit, this means you won't have to change the type of `rmDir` here, so those extra `force` arguments you added elsewhere can go away.

Reply:
Do we need rmDir though @beyackle as it deletes just an empty directory. What if we change the function to removeDirectory and it accepts an options object like 
```
{
  recurse: true,
  force: true
}
```
and implement stuff based on the options object

Reply:
will only use rimraf here to implement rmDir. I am adding some test cases to this pr.

Reply:
@liweitian i think what Ben is suggesting here is to explicitly add a interface for rimraf

Reply:
add one more interface rmrfDir